plugin/core/sessions.py

```diff
@@ -257,6 +257,56 @@ def get_initialize_params(variables: Dict[str, str], workspace_folders: List[Wor
             },
             "codeLens": {
                 "dynamicRegistration": True
+            },
+            "semanticTokens": {
+                "dynamicRegistration": True,
+                "tokenTypes": [
+                    "namespace",
+                    "type",
+                    "class",
+                    "enum",
+                    "interface",
+                    "struct",
+                    "typeParameter",
+                    "parameter",
+                    "variable",
+                    "property",
+                    "enumMember",
+                    "event",
+                    "function",
+                    "method",
+                    "macro",
+                    "keyword",
+                    "modifier",
+                    "comment",
+                    "string",
+                    "number",
+                    "regexp",
+                    "operator"
+                ],
+                "tokenModifiers": [
+                    "declaration",
+                    "definition",
+                    "readonly",
+                    "static",
+                    "deprecated",
+                    "abstract",
+                    "async",
+                    "modification",
+                    "documentation",
+                    "defaultLibrary"
+                ],
+                "formats": [
+                    "relative"
+                ],
+                "requests": {
+                    "range": True,
+                    "full": {
+                        "delta": True
+                    }
+                },
+                "multilineTokenSupport": False,
+                "overlappingTokenSupport": False
```

Comment:
Wondering what does this mean 🤔 

```js
	/**
	 * Whether the client supports tokens that can overlap each other.
	 */
	overlappingTokenSupport?: boolean;
```

Reply:
Maybe this could be used for things like tags, parameters etc. within doc comments (for JavaDoc/JSDoc highlighting). So the whole doc comment block would get the "comment" token type with "documentation" modifier, and the tags within that block get another token added to that. Not sure if we handle this correctly, I guess one region would just override the other if they are overlapping. This is probably something where the initializing order for the region keys will be important again. But I guess we could set this parameter to `True`.